### PR TITLE
Validation errors not displaying for internal subscription management

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?188686

@dannyroberts / @proteusvacuum

### CODE REVIEW

**Comment by proteusvacuum** [comment]
:+1: 
not a huge fan of single-word commit messages though. makes the log a little cryptic down the line. (I'm also guilty of that often, but figured I'd raise it anyway)

**Comment by orangejenny** [comment]
fair enough

File: corehq/apps/domain/views.py [hunk 1]
```diff
@@ -1255,12 +1255,10 @@ def create_form(form_class):
     @property
     @memoized
     def select_subscription_type_form(self):
-        if self.request.method == 'POST':
-            for form_slug in self.slug_to_form:
```
Comment (dannyroberts) [inline corehq/apps/domain/views.py]: Do you know why this was being called in a loop, and does removing that break something else?

Comment (orangejenny) [inline corehq/apps/domain/views.py]: Looked like the previous code looped through all possible forms and returned when it found a matching key in the request, but the request didn't contain any of those keys, and instead uses a 'slug' key. Tested out saving the form successfully and attempting to save it with errors.

Comment (dannyroberts) [inline corehq/apps/domain/views.py]: OK, just checking. It's always weird to stumble upon such "how did this ever work..." kind of code.

Comment (orangejenny) [inline corehq/apps/domain/views.py]: yeah, this is one where I wonder if it ever worked, given that it's validation on an internal page with few users, might not have gotten heavy qa and wouldn't often be tripped by accident

**Pull request merged**